lib/msf/base/config.rb

```diff
@@ -50,6 +50,24 @@ def self.get_config_root
     end
   end
 
+  # Determines the git hash for this source tree
+  #
+  # @return [String] the git hash for this source tree
+  def self.get_git_hash
+    msfrelease = File.join(InstallRoot, 'msfrelease.txt')
+    hash = 'unknown'
+    @git_hash ||= begin
+      if File.exist?(msfrelease)
+        hash = File.read(msfrelease)
+      else
+        if File.exist?(File.join(InstallRoot, '.git'))
+          hash = `git rev-parse HEAD`
```

Comment:
Also, note that our gemspec also assumes git is in the path already - not saying two wrongs make a right :)